gradunwarp/core/utils.py

```diff
@@ -54,6 +54,12 @@ def transform_coordinates(A, M):
     B1, B2, B3 = _transform_coordinates(A1, A2, A3, M)
     return CoordsVector(B1, B2, B3)
 
+from nibabel.affines import apply_affine
+
+def transform_coordinates(A, M):
+    vecs = np.asanyarray([A.x,A.y,A.z]).T
+    vecs_trans = apply_affine(M, vecs)
+    return CoordsVector(vecs_trans[...,0].T, vecs_trans[...,1].T, vecs_trans[...,2].T)
```

Comment:
This can be done more simply:

```suggestion
    transformed = apply_affine(M, np.stack(A).T).T
    return CoordsVector(*transformed)
```